src/core/auth/google_verifier.go

```diff
@@ -0,0 +1,86 @@
+package google_verifer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type Google_verifier struct {
+	Url string
+}
+
+func New(url string) Google_verifier {
+	googleV := Google_verifier{url}
+	return googleV
+}
+
+func (g Google_verifier) Authenticate(ctx context.Context) (context.Context, error) {
```

Comment:
Similar to above, you should probably define this function for a `Google_verifier` pointer, same with all other functions.